src/qbe_ir_generator.cpp

```diff
@@ -123,13 +123,37 @@ auto
 
 void QbeIrGenerator::Visit(const DeclNode& decl) {
   int id_num = NextLocalNum();
-  WriteInstr_("{} =l alloc4 4", FuncScopeTemp{id_num});
+  if (decl.type.is_ptr) {
+    // QBE doesn't have pointer type, pointer type is represented as 64 bits
+    // integer. Thus, we allocate 8 * 8 bits of space for pointers.
+    // Reference: https://c9x.me/compile/doc/il-v1.2.html#Types
+    WriteInstr_("{} =l alloc8 8", FuncScopeTemp{id_num});
+  } else {
+    WriteInstr_("{} =l alloc4 4", FuncScopeTemp{id_num});
+  }
 
   if (decl.init) {
     decl.init->Accept(*this);
     int init_num = num_recorder.NumOfPrevExpr();
-    WriteInstr_("storew {}, {}", FuncScopeTemp{init_num},
-                FuncScopeTemp{id_num});
+    // A pointer declaration may have two options for its right hand side:
+    if (decl.init->type.is_ptr) {
+      const auto* unary_expr = dynamic_cast<UnaryExprNode*>((decl.init).get());
+      // 1. int* a = &b; rhs is a reference of integer. We need to store b's
+      // address to a, where we need to map b's reg_num back to its id_num.
+      if (unary_expr != nullptr) {
```

Comment:
Since the use of `dynamic_cast` is only to test whether `decl.int` has the type of `UnaryExprNode*`, consider removing the intermediate variable `unary_expr`. At first glance, I thought it's going to be used somewhere else, as its lifetime is also unnecessarily long.
```suggestion
      // 1. int* a = &b; rhs is a reference of integer. We need to store b's
      // address to a, where we need to map b's reg_num back to its id_num.
      if (dynamic_cast<UnaryExprNode*>((decl.init).get())) {
```

Reply:
Great suggestion! Fixed!